impl/src/main/java/org/jboss/weld/bean/proxy/ProxyFactory.java

```diff
@@ -499,7 +500,10 @@ private Class<T> createProxyClass(Class<?> originalClass, String proxyClassName)
 
     private ClassFile newClassFile(String name, int accessFlags, String superclass, String... interfaces) {
         try {
-            return new ClassFile(name, accessFlags, superclass, interfaces);
+            // We need to use a (non-deprecated) method that avoids instantiating DefaultClassFactory.INSTANCE
+            // If that happens, we will have module accessibility issues and the need to use --add-opens clausules
+            // NOTE: the CL and ClassFactory are never really used to define the class, see WeldDefaultProxyServices
+            return new ClassFile(name, accessFlags, superclass, ProxyFactory.class.getClassLoader(), WeldDefaultProxyServices.DummyClassFactoryImpl.INSTANCE, interfaces);
```

Comment:
I am not sure `ProxyFactory.class.getClassLoader()` is the best choice here but we need literally *any* non-null CL impl - it never gets used anyway.
We could use TCCL as well but I know that one can be set to `null` in some cases. I am open to better ideas :)